src/overlays/actors/ovl_Bg_Spot15_Rrbox/z_bg_spot15_rrbox.c

```diff
@@ -21,45 +34,340 @@ const ActorInit Bg_Spot15_Rrbox_InitVars = {
+    int i;
+    Vec3f tempVector2;
+    Vec3f tempVector1;
+    Actor* actor = &this->dyna.actor;
+    f32 yIntersect;
+    f32 returnValue = -32000.0f;
+    s32 bgId;
+
+    func_808B3A34(this);
+    for (i = 0; i < ARRAY_COUNT(D_808B45DC); i++) {
+        tempVector1.x = D_808B45DC[i].x * (actor->scale.x * 10.0f);
+        tempVector1.y = D_808B45DC[i].y * (actor->scale.y * 10.0f);
+        tempVector1.z = D_808B45DC[i].z * (actor->scale.z * 10.0f);
+
+        func_808B39E8(&tempVector2, &tempVector1, this->unk_16C, this->unk_170);
+
+        tempVector2.x += actor->posRot.pos.x;
+        tempVector2.y += actor->pos4.y;
+        tempVector2.z += actor->posRot.pos.z;
+
+        yIntersect = func_8003CA64(&globalCtx->colCtx, &actor->floorPoly, &bgId, actor, &tempVector2, 0);
+
+        if (returnValue < yIntersect) {
+            returnValue = yIntersect;
+            this->bgId = bgId;
+        }
+    }
+    return returnValue;
+}
+
+s32 func_808B3F58(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    if (func_808B3CA0(this, globalCtx, 0)) {
+        return true;
+    }
+    if (func_808B3CA0(this, globalCtx, 1)) {
+        return true;
+    }
+    if (func_808B3CA0(this, globalCtx, 2)) {
+        return true;
+    }
+    if (func_808B3CA0(this, globalCtx, 3)) {
+        return true;
+    }
+    if (func_808B3CA0(this, globalCtx, 4)) {
+        return true;
+    }
+    return false;
+}
+
+s32 func_808B4010(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    return func_800435D8(globalCtx, &this->dyna, this->dyna.actor.scale.x * 290.0f,
+                         this->dyna.actor.scale.x * 290.0f + 20.0f, 1.0f) == 0;
+}
+
+void func_808B4084(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    this->actionFunc = func_808B40AC;
+    this->dyna.actor.gravity = 0.0f;
+    this->dyna.actor.velocity.x = 0.0f;
+    this->dyna.actor.velocity.y = 0.0f;
+    this->dyna.actor.velocity.z = 0.0f;
+}
+
+void func_808B40AC(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (this->unk_168 <= 0 && fabsf(this->dyna.unk_150) > 0.001f) {
+        if (func_808B3AAC(this, globalCtx) != 0 && func_808B4010(this, globalCtx) == 0) {
+            this->unk_17C = this->dyna.unk_150;
+            func_808B4178(this, globalCtx);
+        }
+        else {
+            player->stateFlags2 &= ~0x10;
+            this->dyna.unk_150 = 0.0f;
+        }
+    }
+    else {
+        player->stateFlags2 &= ~0x10;
+        this->dyna.unk_150 = 0.0f;
+    }
+}
+
+void func_808B4178(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    this->actionFunc = func_808B4194;
+    this->dyna.actor.gravity = 0.0f;
+}
+
+void func_808B4194(BgSpot15Rrbox* this, GlobalContext* globalCtx) {
+    f32 sign;
+    Player* player = PLAYER;
+    f32 temp_f0_2;
+    s32 approxFResult;
+
+    Actor* actor = &this->dyna.actor;
+
+    this->unk_174 = this->unk_174 + 0.5f;
+    if (2.0f < this->unk_174) {
```

Comment:
constants should be on the right side of conditions like this unless it's impossible for some reason